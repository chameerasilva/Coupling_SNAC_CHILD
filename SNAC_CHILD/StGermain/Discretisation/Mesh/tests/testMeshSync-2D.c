/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Patrick D. Sunter, Software Engineer, VPAC. (pds@vpac.org)
**	Luke J. Hodkinson, Computational Engineer, VPAC. (lhodkins@vpac.org)
**	Siew-Ching Tan, Software Engineer, VPAC. (siew@vpac.org)
**	Alan H. Lo, Computational Engineer, VPAC. (alan@vpac.org)
**	Raquibul Hassan, Computational Engineer, VPAC. (raq@vpac.org)
**
**  This library is free software; you can redistribute it and/or
**  modify it under the terms of the GNU Lesser General Public
**  License as published by the Free Software Foundation; either
**  version 2.1 of the License, or (at your option) any later version.
**
**  This library is distributed in the hope that it will be useful,
**  but WITHOUT ANY WARRANTY; without even the implied warranty of
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
**  Lesser General Public License for more details.
**
**  You should have received a copy of the GNU Lesser General Public
**  License along with this library; if not, write to the Free Software
**  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
**
** $Id: testMeshSync.c 2136 2004-09-30 02:47:13Z PatrickSunter $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include "Base/Base.h"

#include "Discretisation/Geometry/Geometry.h"
#include "Discretisation/Shape/Shape.h"
#include "Discretisation/Mesh/Mesh.h"

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <mpi.h>

struct _Node
{
	double temp;
};

struct _Element
{
	double temp;
};


int main( int argc, char* argv[] )
{
	MPI_Comm		CommWorld;
	int			rank;
	int			procCount;
	int			procToWatch;
	Dictionary*		dictionary;
	ExtensionManager_Register*	extensionMgr_Register;
	Topology*		nTopology;
	ElementLayout*		eLayout;
	NodeLayout*		nLayout;
	MeshDecomp*		decomp;
	MeshLayout*		ml;
	Mesh*			mesh;
	Stream*			stream;
	Node_DomainIndex	node_dI;
	Element_DomainIndex	element_dI;
	
	/* Initialise MPI, get world info */
	MPI_Init(&argc, &argv);
	MPI_Comm_dup( MPI_COMM_WORLD, &CommWorld );
	MPI_Comm_size(CommWorld, &procCount);
	MPI_Comm_rank(CommWorld, &rank);

	Base_Init( &argc, &argv );
	
	DiscretisationGeometry_Init( &argc, &argv );
	DiscretisationShape_Init( &argc, &argv );
	DiscretisationMesh_Init( &argc, &argv );
	MPI_Barrier( CommWorld ); /* Ensures copyright info always come first in output */

	stream = Journal_Register (Info_Type, "myStream");
	procToWatch = argc >= 2 ? atoi(argv[1]) : 0;
	
	dictionary = Dictionary_New();
	Dictionary_Add( dictionary, "rank", Dictionary_Entry_Value_FromUnsignedInt( rank ) );
	Dictionary_Add( dictionary, "numProcessors", Dictionary_Entry_Value_FromUnsignedInt( procCount ) );
	Dictionary_Add( dictionary, "meshSizeI", Dictionary_Entry_Value_FromUnsignedInt( 3 ) );
	Dictionary_Add( dictionary, "meshSizeJ", Dictionary_Entry_Value_FromUnsignedInt( 3 ) );
	Dictionary_Add( dictionary, "meshSizeK", Dictionary_Entry_Value_FromUnsignedInt( 3 ) );
	Dictionary_Add( dictionary, "allowUnusedCPUs", Dictionary_Entry_Value_FromBool( False ) );
	Dictionary_Add( dictionary, "allowPartitionOnElement", Dictionary_Entry_Value_FromBool( False ) );
	Dictionary_Add( dictionary, "allowPartitionOnNode", Dictionary_Entry_Value_FromBool( True ) );
	Dictionary_Add( dictionary, "allowUnbalancing", Dictionary_Entry_Value_FromBool( False ) );
	Dictionary_Add( dictionary, "shadowDepth", Dictionary_Entry_Value_FromUnsignedInt( 1 ) );
	
	nTopology = (Topology*)IJK6Topology_New( "IJK6Topology", dictionary );
	eLayout = (ElementLayout*)ParallelPipedHexaEL_New( "PPHexaEL", 3, dictionary );
	nLayout = (NodeLayout*)CornerNL_New( "CornerNL", dictionary, eLayout, nTopology );
	decomp = (MeshDecomp*)HexaMD_New_All( "HexaMD", dictionary, MPI_COMM_WORLD, eLayout, nLayout, 2 );
	ml = MeshLayout_New( "MeshLayout", eLayout, nLayout, decomp );
	
	extensionMgr_Register = ExtensionManager_Register_New();
	mesh = Mesh_New( "Mesh", ml, sizeof(Node), sizeof(Element), extensionMgr_Register, dictionary );
	
	mesh->buildNodeLocalToGlobalMap = True;
	mesh->buildNodeDomainToGlobalMap = True;
	mesh->buildNodeGlobalToLocalMap = True;
	mesh->buildNodeGlobalToDomainMap = True;
	mesh->buildNodeNeighbourTbl = True;
	mesh->buildNodeElementTbl = True;
	mesh->buildElementLocalToGlobalMap = True;
	mesh->buildElementDomainToGlobalMap = True;
	mesh->buildElementGlobalToDomainMap = True;
	mesh->buildElementGlobalToLocalMap = True;
	mesh->buildElementNeighbourTbl = True;
	mesh->buildElementNodeTbl = True;
	Build( mesh, 0, False );
	Initialise(mesh, 0, False );
	
	for( node_dI = 0; node_dI < mesh->nodeDomainCount; node_dI++ ) {
		mesh->nodeCoord[node_dI][0] = (double)rank;
		mesh->nodeCoord[node_dI][1] = (double)rank;
		mesh->nodeCoord[node_dI][2] = (double)rank;
		mesh->node[node_dI].temp = 10.0 + (double)rank;
	}
	for( element_dI = 0; element_dI < mesh->elementDomainCount; element_dI++ ) {
		mesh->element[element_dI].temp = 20.0 + (double)rank;
	}
	
	if( rank == procToWatch ) {
		Print(mesh, stream);
		
		printf( "Pre-sync:\n" );
		for( node_dI = 0; node_dI < mesh->nodeDomainCount; node_dI++ ) {
			printf( "\tnodeCoord[%d]: { %g, %g, %g } - temp: %g\n",  node_dI, mesh->nodeCoord[node_dI][0],
				mesh->nodeCoord[node_dI][1], mesh->nodeCoord[node_dI][2], mesh->node[node_dI].temp );
		}
		for( element_dI = 0; element_dI < mesh->elementDomainCount; element_dI++ )
			printf( "\telement[%d]: temp: %g\n", element_dI, mesh->element[element_dI].temp );
	}
	
	Mesh_Sync( mesh );
	
	if( rank == procToWatch ) {
		printf( "Post-sync:\n" );
		for( node_dI = 0; node_dI < mesh->nodeDomainCount; node_dI++ ) {
			printf( "\tnodeCoord[%d]: { %g, %g, %g } - temp: %g\n",  node_dI, mesh->nodeCoord[node_dI][0],
				mesh->nodeCoord[node_dI][1], mesh->nodeCoord[node_dI][2], mesh->node[node_dI].temp );
		}
		for( element_dI = 0; element_dI < mesh->elementDomainCount; element_dI++ )
			printf( "\telement[%d]: temp: %g\n", element_dI, mesh->element[element_dI].temp );
	}
	
	Stg_Class_Delete( mesh );
	Stg_Class_Delete( ml );
	Stg_Class_Delete( decomp );
	Stg_Class_Delete( nLayout );
	Stg_Class_Delete( eLayout );
	Stg_Class_Delete( nTopology );
	Stg_Class_Delete( dictionary );
	
	DiscretisationMesh_Finalise();
	DiscretisationShape_Finalise();
	DiscretisationGeometry_Finalise();
	
	Base_Finalise();
	
	/* Close off MPI */
	MPI_Finalize();
	
	return 0; /* success */
}
