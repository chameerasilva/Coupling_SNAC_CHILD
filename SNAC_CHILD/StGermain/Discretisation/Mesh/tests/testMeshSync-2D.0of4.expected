StGermain Framework revision 3605. Copyright (C) 2003-2005 VPAC.
Mesh (ptr): (0x805f770)
Stg_Component (ptr): 0x805f770
Stg_Object (ptr): 0x805f770
	Stg_Class (ptr): 0x805f770
		sizeOfSelf: 400
		_deleteSelf: Yes
		type: Mesh
		_delete (func ptr): 0xb7efe1dd
		_print (func ptr): 0xb7efea6a
		_copy (func ptr): 0xb7effc4e
	name: Mesh
	nameAllocationType: NON_GLOBAL
	_defaultConstructor (func ptr): 0xb7efd51c
	_construct (func ptr): 0xb7efe00e
	_build (func ptr): 0xb7f00322
	_initialise (func ptr): 0xb7f01214
	_execute (func ptr): 0xb7f01219
	_destroy (func ptr): 0xb7efe1d8
	isConstructed: True
	isBuilt: True
	isInitialised: True
	hasExecuted: False
	isDestroyed: False
	constructor function name: Mesh-Construct
	build function name: Mesh-Build
	initialise function name: Mesh-Initialise
	execute function name: Mesh-Execute
	destroy function name: Mesh-Destroy
	dictionary (ptr): 0x805c5a8
	layout (ptr): 0x805f4c0
Extension (ptr):0x805fb40
Stg_Object (ptr): 0x805fb40
	Stg_Class (ptr): 0x805fb40
		sizeOfSelf: 80
		_deleteSelf: Yes
		type: Extension
		_delete (func ptr): 0xb7e76018
		_print (func ptr): 0xb7e76066
		_copy (func ptr): 0xb7e76200
	name: node
	nameAllocationType: NON_GLOBAL
	initialSize 8
	finalSize 8
	count: 0
	extInfos (ptr): 0x805fba0
Stg_ObjectList (ptr):0x805fba0
	Stg_Class (ptr): 0x805fba0
		sizeOfSelf: 96
		_deleteSelf: Yes
		type: Stg_ObjectList
		_delete (func ptr): 0xb7ea7f3e
		_print (func ptr): 0xb7ea7f75
		_copy (func ptr): 0xb7ea8217
	_append(func ptr): 0xb7ea8e7b
	_prepend(func ptr): 0xb7ea8f41
	_replaceAll(func ptr): 0xb7ea9005
	_replace(func ptr): 0xb7ea90be
	_insertBefore(func ptr): 0xb7ea925b
	_insertAfter(func ptr): 0xb7ea9416
	_remove(func ptr): 0xb7ea95d3
	_getIndex(func ptr): 0xb7ea9679
	_get(func ptr): 0xb7ea9727
	_allocMoreMemory(func ptr): 0xb7ea98fa
	_insertAtIndex(func ptr): 0xb7ea9980
	_removeByIndex(func ptr): 0xb7ea99fe
	_deleteAllObjects(func ptr): 0xb7ea987b
	_size: 8
	_delta: 8
	_noJournalingInCopy: 0
	count: 0
	data[0-0]:
	_extensionsToExisting (ptr): (nil)
	_existingObject (ptr): (nil)
	_array (ptr): (nil)
	itemSize 0
	em (ptr) (nil)
	count 0
Extension (ptr):0x805fc10
Stg_Object (ptr): 0x805fc10
	Stg_Class (ptr): 0x805fc10
		sizeOfSelf: 80
		_deleteSelf: Yes
		type: Extension
		_delete (func ptr): 0xb7e76018
		_print (func ptr): 0xb7e76066
		_copy (func ptr): 0xb7e76200
	name: element
	nameAllocationType: NON_GLOBAL
	initialSize 8
	finalSize 8
	count: 0
	extInfos (ptr): 0x805fc70
Stg_ObjectList (ptr):0x805fc70
	Stg_Class (ptr): 0x805fc70
		sizeOfSelf: 96
		_deleteSelf: Yes
		type: Stg_ObjectList
		_delete (func ptr): 0xb7ea7f3e
		_print (func ptr): 0xb7ea7f75
		_copy (func ptr): 0xb7ea8217
	_append(func ptr): 0xb7ea8e7b
	_prepend(func ptr): 0xb7ea8f41
	_replaceAll(func ptr): 0xb7ea9005
	_replace(func ptr): 0xb7ea90be
	_insertBefore(func ptr): 0xb7ea925b
	_insertAfter(func ptr): 0xb7ea9416
	_remove(func ptr): 0xb7ea95d3
	_getIndex(func ptr): 0xb7ea9679
	_get(func ptr): 0xb7ea9727
	_allocMoreMemory(func ptr): 0xb7ea98fa
	_insertAtIndex(func ptr): 0xb7ea9980
	_removeByIndex(func ptr): 0xb7ea99fe
	_deleteAllObjects(func ptr): 0xb7ea987b
	_size: 8
	_delta: 8
	_noJournalingInCopy: 0
	count: 0
	data[0-0]:
	_extensionsToExisting (ptr): (nil)
	_existingObject (ptr): (nil)
	_array (ptr): (nil)
	itemSize 0
	em (ptr) (nil)
	count 0
	node (ptr): 0x80604a0
	nodeLocalCount: 12
	nodeDomainCount: 27
	nodeShadowCount: 15
	nodeGlobalCount: 27
	nodeCoord (ptr): 0x8060588
	nodeCoord[0-27]:   (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), (0.00,0.00,0.00), 
	ownsNodeLocalToGlobalMap: 1
	nodeL2G (ptr): 0x8060280
	nodeL2G[0-12]:   000 001 003 004 009 010 012 013 018 019 021 022 
	ownsNodeDomainToGlobalMap: 1
	nodeD2G (ptr): 0x8060280
	nodeD2G[0-27]:   000 001 003 004 009 010 012 013 018 019 021 022 002 005 006 007 008 011 014 015 016 017 020 023 024 025 026 
	ownsNodeGlobalToLocalMap: 1
	nodeG2L (ptr): 0x8060898
	nodeG2L[0-27]:   000 001 027 002 003 027 027 027 027 004 005 027 006 007 027 027 027 027 008 009 027 010 011 027 027 027 027 
	ownsNodeGlobalToDomainMap: 1
	nodeG2D (ptr): 0x8060820
	nodeG2D[0-27]:   000 001 012 002 003 013 014 015 016 004 005 017 006 007 018 019 020 021 008 009 022 010 011 023 024 025 026 
	ownsNodeNeighbourCountTbl: 1
	nodeNeighbourCountTbl (ptr): 0x8060910
	ownsNodeNeighbourTbl: 1
	nodeNeighbourTbl (ptr): 0x8060988
	nodeNeighbourTbl[0-27]:
		nodeNeighbourTbl[0][0-6]: 001 002 004 027 027 027 
		nodeNeighbourTbl[1][0-6]: 012 003 005 000 027 027 
		nodeNeighbourTbl[2][0-6]: 003 014 006 027 000 027 
		nodeNeighbourTbl[3][0-6]: 013 015 007 002 001 027 
		nodeNeighbourTbl[4][0-6]: 005 006 008 027 027 000 
		nodeNeighbourTbl[5][0-6]: 017 007 009 004 027 001 
		nodeNeighbourTbl[6][0-6]: 007 019 010 027 004 002 
		nodeNeighbourTbl[7][0-6]: 018 020 011 006 005 003 
		nodeNeighbourTbl[8][0-6]: 009 010 027 027 027 004 
		nodeNeighbourTbl[9][0-6]: 022 011 027 008 027 005 
		nodeNeighbourTbl[10][0-6]: 011 024 027 027 008 006 
		nodeNeighbourTbl[11][0-6]: 023 025 027 010 009 007 
		nodeNeighbourTbl[12][0-6]: 027 013 017 001 027 027 
		nodeNeighbourTbl[13][0-6]: 027 016 018 003 012 027 
		nodeNeighbourTbl[14][0-6]: 015 027 019 027 002 027 
		nodeNeighbourTbl[15][0-6]: 016 027 020 014 003 027 
		nodeNeighbourTbl[16][0-6]: 027 027 021 015 013 027 
		nodeNeighbourTbl[17][0-6]: 027 018 022 005 027 012 
		nodeNeighbourTbl[18][0-6]: 027 021 023 007 017 013 
		nodeNeighbourTbl[19][0-6]: 020 027 024 027 006 014 
		nodeNeighbourTbl[20][0-6]: 021 027 025 019 007 015 
		nodeNeighbourTbl[21][0-6]: 027 027 026 020 018 016 
		nodeNeighbourTbl[22][0-6]: 027 023 027 009 027 017 
		nodeNeighbourTbl[23][0-6]: 027 026 027 011 022 018 
		nodeNeighbourTbl[24][0-6]: 025 027 027 027 010 019 
		nodeNeighbourTbl[25][0-6]: 026 027 027 024 011 020 
		nodeNeighbourTbl[26][0-6]: 027 027 027 025 023 021 
	ownsNodeElementCountTbl: 1
	nodeElementCountTbl (ptr): 0x8060c88
	ownsNodeElementTbl: 1
	nodeElementTbl (ptr): 0x8060d00
	nodeElementTbl[0-27]:
		nodeElementTbl[0][0-8]: 008 008 008 008 008 008 008 000 
		nodeElementTbl[1][0-8]: 008 008 008 008 008 008 000 002 
		nodeElementTbl[2][0-8]: 008 008 008 008 008 000 008 003 
		nodeElementTbl[3][0-8]: 008 008 008 008 000 002 003 004 
		nodeElementTbl[4][0-8]: 008 008 008 000 008 008 008 001 
		nodeElementTbl[5][0-8]: 008 008 000 002 008 008 001 005 
		nodeElementTbl[6][0-8]: 008 000 008 003 008 001 008 006 
		nodeElementTbl[7][0-8]: 000 002 003 004 001 005 006 007 
		nodeElementTbl[8][0-8]: 008 008 008 001 008 008 008 008 
		nodeElementTbl[9][0-8]: 008 008 001 005 008 008 008 008 
		nodeElementTbl[10][0-8]: 008 001 008 006 008 008 008 008 
		nodeElementTbl[11][0-8]: 001 005 006 007 008 008 008 008 
		nodeElementTbl[12][0-8]: 008 008 008 008 008 008 002 008 
		nodeElementTbl[13][0-8]: 008 008 008 008 002 008 004 008 
		nodeElementTbl[14][0-8]: 008 008 008 008 008 003 008 008 
		nodeElementTbl[15][0-8]: 008 008 008 008 003 004 008 008 
		nodeElementTbl[16][0-8]: 008 008 008 008 004 008 008 008 
		nodeElementTbl[17][0-8]: 008 008 002 008 008 008 005 008 
		nodeElementTbl[18][0-8]: 002 008 004 008 005 008 007 008 
		nodeElementTbl[19][0-8]: 008 003 008 008 008 006 008 008 
		nodeElementTbl[20][0-8]: 003 004 008 008 006 007 008 008 
		nodeElementTbl[21][0-8]: 004 008 008 008 007 008 008 008 
		nodeElementTbl[22][0-8]: 008 008 005 008 008 008 008 008 
		nodeElementTbl[23][0-8]: 005 008 007 008 008 008 008 008 
		nodeElementTbl[24][0-8]: 008 006 008 008 008 008 008 008 
		nodeElementTbl[25][0-8]: 006 007 008 008 008 008 008 008 
		nodeElementTbl[26][0-8]: 007 008 008 008 008 008 008 008 
	element (ptr): 0x80610d8
	elementLocalCount: 2
	elementDomainCount: 8
	elementShadowCount: 6
	elementGlobalCount: 8
	ownsElementLocalToGlobalMap: 1
	elementL2G (ptr): 0x805d048
	elementL2G[0-2]:   000 004 
	ownsElementDomainToGlobalMap: 1
	elementD2G (ptr): 0x805d048
	elementD2G[0-8]:   000 004 001 002 003 005 006 007 
	ownsElementGlobalToLocalMap: 1
	elementG2L (ptr): 0x8061158
	elementG2L[0-8]:   000 008 008 008 001 008 008 008 
	ownsElementGlobalToDomainMap: 1
	elementG2D (ptr): 0x8061128
	elementG2D[0-8]:   000 002 003 004 001 005 006 007 
	ownsElementNeighbourCountTbl: 1
	elementNeighbourCountTbl (ptr): 0x8061188
	ownsElementNeighbourTbl: 1
	elementNeighbourTbl (ptr): 0x80611b8
	elementNeighbourTbl[0-8]:
		elementNeighbourTbl[0][0-26]: 008 008 008 008 008 008 008 008 008 008 008 008 008 002 008 003 004 008 008 008 008 001 005 008 006 007 
		elementNeighbourTbl[1][0-26]: 008 008 008 008 000 002 008 003 004 008 008 008 008 005 008 006 007 008 008 008 008 008 008 008 008 008 
		elementNeighbourTbl[2][0-26]: 008 008 008 008 008 008 008 008 008 008 008 008 000 008 003 004 008 008 008 008 001 005 008 006 007 008 
		elementNeighbourTbl[3][0-26]: 008 008 008 008 008 008 008 008 008 008 000 002 008 004 008 008 008 008 001 005 008 006 007 008 008 008 
		elementNeighbourTbl[4][0-26]: 008 008 008 008 008 008 008 008 008 000 002 008 003 008 008 008 008 001 005 008 006 007 008 008 008 008 
		elementNeighbourTbl[5][0-26]: 008 008 008 000 002 008 003 004 008 008 008 008 001 008 006 007 008 008 008 008 008 008 008 008 008 008 
		elementNeighbourTbl[6][0-26]: 008 000 002 008 003 004 008 008 008 008 001 005 008 007 008 008 008 008 008 008 008 008 008 008 008 008 
		elementNeighbourTbl[7][0-26]: 000 002 008 003 004 008 008 008 008 001 005 008 006 008 008 008 008 008 008 008 008 008 008 008 008 008 
	ownsElementNodeCountTbl: 1
	elementNodeCountTbl (ptr): 0x805aa50
	ownsElementNodeTbl: 1
	elementNodeTbl (ptr): 0x8061528
	elementNodeTbl[0-8]:
		elementNodeTbl[0][0-8]: 000 001 003 002 004 005 007 006 
		elementNodeTbl[1][0-8]: 004 005 007 006 008 009 011 010 
		elementNodeTbl[2][0-8]: 001 012 013 003 005 017 018 007 
		elementNodeTbl[3][0-8]: 002 003 015 014 006 007 020 019 
		elementNodeTbl[4][0-8]: 003 013 016 015 007 018 021 020 
		elementNodeTbl[5][0-8]: 005 017 018 007 009 022 023 011 
		elementNodeTbl[6][0-8]: 006 007 020 019 010 011 025 024 
		elementNodeTbl[7][0-8]: 007 018 021 020 011 023 026 025 
	procNbrCnt: 3
	procNbrTbl (ptr): 0x80602f8
		procNbrTbl[0]: 1
		procNbrTbl[1]: 2
		procNbrTbl[2]: 3
	elementShadowInfo->procShadowedCnt (ptr): 0x8060468
		elementShadowInfo->procShadowedCnt[0]: 2
		elementShadowInfo->procShadowedCnt[1]: 2
		elementShadowInfo->procShadowedCnt[2]: 2
	elementShadowInfo->procShadowedTbl (ptr): 0x805d078
		elementShadowInfo->procShadowedTbl[0] (ptr): 0x8060388
			elementShadowInfo->procShadowedTbl[0][0]: 0
			elementShadowInfo->procShadowedTbl[0][1]: 1
		elementShadowInfo->procShadowedTbl[1] (ptr): 0x8061750
			elementShadowInfo->procShadowedTbl[1][0]: 0
			elementShadowInfo->procShadowedTbl[1][1]: 1
		elementShadowInfo->procShadowedTbl[2] (ptr): 0x80617a8
			elementShadowInfo->procShadowedTbl[2][0]: 0
			elementShadowInfo->procShadowedTbl[2][1]: 1
	elementShadowInfo->procShadowCnt (ptr): 0x80617c0
		elementShadowInfo->procShadowCnt[0]: 2
		elementShadowInfo->procShadowCnt[1]: 2
		elementShadowInfo->procShadowCnt[2]: 2
	elementShadowInfo->procShadowTbl (ptr): 0x80617d8
		elementShadowInfo->procShadowTbl[0] (ptr): 0x80617f0
			elementShadowInfo->procShadowTbl[0][0]: 2
			elementShadowInfo->procShadowTbl[0][1]: 5
		elementShadowInfo->procShadowTbl[1] (ptr): 0x8061808
			elementShadowInfo->procShadowTbl[1][0]: 3
			elementShadowInfo->procShadowTbl[1][1]: 6
		elementShadowInfo->procShadowTbl[2] (ptr): 0x8061820
			elementShadowInfo->procShadowTbl[2][0]: 4
			elementShadowInfo->procShadowTbl[2][1]: 7
	nodeShadowInfo->procShadowedCnt (ptr): 0x8061878
		nodeShadowInfo->procShadowedCnt[0]: 6
		nodeShadowInfo->procShadowedCnt[1]: 6
		nodeShadowInfo->procShadowedCnt[2]: 9
	nodeShadowInfo->procShadowedTbl (ptr): 0x8061890
		nodeShadowInfo->procShadowedTbl[0] (ptr): 0x80616a8
			nodeShadowInfo->procShadowedTbl[0][0]: 0
			nodeShadowInfo->procShadowedTbl[0][1]: 2
			nodeShadowInfo->procShadowedTbl[0][2]: 4
			nodeShadowInfo->procShadowedTbl[0][3]: 6
			nodeShadowInfo->procShadowedTbl[0][4]: 8
			nodeShadowInfo->procShadowedTbl[0][5]: 10
		nodeShadowInfo->procShadowedTbl[1] (ptr): 0x8061680
			nodeShadowInfo->procShadowedTbl[1][0]: 0
			nodeShadowInfo->procShadowedTbl[1][1]: 1
			nodeShadowInfo->procShadowedTbl[1][2]: 4
			nodeShadowInfo->procShadowedTbl[1][3]: 5
			nodeShadowInfo->procShadowedTbl[1][4]: 8
			nodeShadowInfo->procShadowedTbl[1][5]: 9
		nodeShadowInfo->procShadowedTbl[2] (ptr): 0x8061a70
			nodeShadowInfo->procShadowedTbl[2][0]: 0
			nodeShadowInfo->procShadowedTbl[2][1]: 1
			nodeShadowInfo->procShadowedTbl[2][2]: 2
			nodeShadowInfo->procShadowedTbl[2][3]: 4
			nodeShadowInfo->procShadowedTbl[2][4]: 5
			nodeShadowInfo->procShadowedTbl[2][5]: 6
			nodeShadowInfo->procShadowedTbl[2][6]: 8
			nodeShadowInfo->procShadowedTbl[2][7]: 9
			nodeShadowInfo->procShadowedTbl[2][8]: 10
	nodeShadowInfo->procShadowCnt (ptr): 0x8061aa0
		nodeShadowInfo->procShadowCnt[0]: 6
		nodeShadowInfo->procShadowCnt[1]: 6
		nodeShadowInfo->procShadowCnt[2]: 9
	nodeShadowInfo->procShadowTbl (ptr): 0x8061ab8
		nodeShadowInfo->procShadowTbl[0] (ptr): 0x8061658
			nodeShadowInfo->procShadowTbl[0][0]: 12
			nodeShadowInfo->procShadowTbl[0][1]: 13
			nodeShadowInfo->procShadowTbl[0][2]: 17
			nodeShadowInfo->procShadowTbl[0][3]: 18
			nodeShadowInfo->procShadowTbl[0][4]: 22
			nodeShadowInfo->procShadowTbl[0][5]: 23
		nodeShadowInfo->procShadowTbl[1] (ptr): 0x8061ad0
			nodeShadowInfo->procShadowTbl[1][0]: 14
			nodeShadowInfo->procShadowTbl[1][1]: 15
			nodeShadowInfo->procShadowTbl[1][2]: 19
			nodeShadowInfo->procShadowTbl[1][3]: 20
			nodeShadowInfo->procShadowTbl[1][4]: 24
			nodeShadowInfo->procShadowTbl[1][5]: 25
		nodeShadowInfo->procShadowTbl[2] (ptr): 0x8061af8
			nodeShadowInfo->procShadowTbl[2][0]: 13
			nodeShadowInfo->procShadowTbl[2][1]: 15
			nodeShadowInfo->procShadowTbl[2][2]: 16
			nodeShadowInfo->procShadowTbl[2][3]: 18
			nodeShadowInfo->procShadowTbl[2][4]: 20
			nodeShadowInfo->procShadowTbl[2][5]: 21
			nodeShadowInfo->procShadowTbl[2][6]: 23
			nodeShadowInfo->procShadowTbl[2][7]: 25
			nodeShadowInfo->procShadowTbl[2][8]: 26
Pre-sync:
	nodeCoord[0]: { 0, 0, 0 } - temp: 10
	nodeCoord[1]: { 0, 0, 0 } - temp: 10
	nodeCoord[2]: { 0, 0, 0 } - temp: 10
	nodeCoord[3]: { 0, 0, 0 } - temp: 10
	nodeCoord[4]: { 0, 0, 0 } - temp: 10
	nodeCoord[5]: { 0, 0, 0 } - temp: 10
	nodeCoord[6]: { 0, 0, 0 } - temp: 10
	nodeCoord[7]: { 0, 0, 0 } - temp: 10
	nodeCoord[8]: { 0, 0, 0 } - temp: 10
	nodeCoord[9]: { 0, 0, 0 } - temp: 10
	nodeCoord[10]: { 0, 0, 0 } - temp: 10
	nodeCoord[11]: { 0, 0, 0 } - temp: 10
	nodeCoord[12]: { 0, 0, 0 } - temp: 10
	nodeCoord[13]: { 0, 0, 0 } - temp: 10
	nodeCoord[14]: { 0, 0, 0 } - temp: 10
	nodeCoord[15]: { 0, 0, 0 } - temp: 10
	nodeCoord[16]: { 0, 0, 0 } - temp: 10
	nodeCoord[17]: { 0, 0, 0 } - temp: 10
	nodeCoord[18]: { 0, 0, 0 } - temp: 10
	nodeCoord[19]: { 0, 0, 0 } - temp: 10
	nodeCoord[20]: { 0, 0, 0 } - temp: 10
	nodeCoord[21]: { 0, 0, 0 } - temp: 10
	nodeCoord[22]: { 0, 0, 0 } - temp: 10
	nodeCoord[23]: { 0, 0, 0 } - temp: 10
	nodeCoord[24]: { 0, 0, 0 } - temp: 10
	nodeCoord[25]: { 0, 0, 0 } - temp: 10
	nodeCoord[26]: { 0, 0, 0 } - temp: 10
	element[0]: temp: 20
	element[1]: temp: 20
	element[2]: temp: 20
	element[3]: temp: 20
	element[4]: temp: 20
	element[5]: temp: 20
	element[6]: temp: 20
	element[7]: temp: 20
Post-sync:
	nodeCoord[0]: { 0, 0, 0 } - temp: 10
	nodeCoord[1]: { 0, 0, 0 } - temp: 10
	nodeCoord[2]: { 0, 0, 0 } - temp: 10
	nodeCoord[3]: { 0, 0, 0 } - temp: 10
	nodeCoord[4]: { 0, 0, 0 } - temp: 10
	nodeCoord[5]: { 0, 0, 0 } - temp: 10
	nodeCoord[6]: { 0, 0, 0 } - temp: 10
	nodeCoord[7]: { 0, 0, 0 } - temp: 10
	nodeCoord[8]: { 0, 0, 0 } - temp: 10
	nodeCoord[9]: { 0, 0, 0 } - temp: 10
	nodeCoord[10]: { 0, 0, 0 } - temp: 10
	nodeCoord[11]: { 0, 0, 0 } - temp: 10
	nodeCoord[12]: { 1, 1, 1 } - temp: 11
	nodeCoord[13]: { 3, 3, 3 } - temp: 13
	nodeCoord[14]: { 2, 2, 2 } - temp: 12
	nodeCoord[15]: { 3, 3, 3 } - temp: 13
	nodeCoord[16]: { 3, 3, 3 } - temp: 13
	nodeCoord[17]: { 1, 1, 1 } - temp: 11
	nodeCoord[18]: { 3, 3, 3 } - temp: 13
	nodeCoord[19]: { 2, 2, 2 } - temp: 12
	nodeCoord[20]: { 3, 3, 3 } - temp: 13
	nodeCoord[21]: { 3, 3, 3 } - temp: 13
	nodeCoord[22]: { 1, 1, 1 } - temp: 11
	nodeCoord[23]: { 3, 3, 3 } - temp: 13
	nodeCoord[24]: { 2, 2, 2 } - temp: 12
	nodeCoord[25]: { 3, 3, 3 } - temp: 13
	nodeCoord[26]: { 3, 3, 3 } - temp: 13
	element[0]: temp: 20
	element[1]: temp: 20
	element[2]: temp: 21
	element[3]: temp: 22
	element[4]: temp: 23
	element[5]: temp: 21
	element[6]: temp: 22
	element[7]: temp: 23
